docs/how-to-upload-manifest.md

```diff
@@ -1,107 +1,92 @@
-$ chmod a+x login.sh
-$ ./login.sh {username} {password}
+$ ./scripts/get-quay-token
 ```
 
-Upon successful invocation, you will receive a JSON response with an access token, as shown below. Save the value of the `token` field from the JSON response. We will use this token to make calls to `quay.io` API. 
+You will be prompted to provide your `quay.io` account user name and password.
+
+Upon successful invocation, you will receive a JSON response with an access token, as shown below. Save the value of the `token` field from the JSON response. We will use this token to make calls to `quay.io` API.
+
 ```json
 {
     "token": "basic XWtgc2hlbTpsZWR6ZXCwbYlf"
 }
 ```
 
 ## Push Operator Manifest
-* Save the manifest into a file, for example `myapp.yaml`
-* Copy the following excerpt and save it into a file, for example `push.sh`. Put appropriate values for the variables.
+
+You can now push the manifests to quay by using the following bash script:
 
 ```bash
-#!/bin/sh
+$ ./scripts/push-to-quay
+```
 
-FILENAME="file name (without the extension)"
-NAMESPACE="namespace in quay.io"
-REPOSITORY="repository name in quay.io"
-RELEASE="version/release of the operator"
-TOKEN="basic {access token here}"
+You will be prompted for the path to the file containing your operator manifest, the namespace, repo, release and token related to your quay registry.
 
-function cleanup() {
-    rm -f ${FILENAME}.tar.gz
-}
-trap cleanup EXIT
+For example, if you have saved the operator manifest into `myapp.yaml` and want to push it to `myoperators/myapp:1.0.0` then the variables will have the following values.
 
-tar czf ${FILENAME}.tar.gz ${FILENAME}.yaml
+    file: myapp.yaml
+    namespace: myoperators
+    repository: myapp
+    release: 1.0.0
+    token: basic XWtgc2hlbTpsZWR6ZXCwbYlf
 
-BLOB=$(cat ${FILENAME}.tar.gz | base64 -w 0)
+You should now be able to visit `quay.io` and browse the operator manifest in your desired namespace.
 
-curl -H "Content-Type: application/json" \
-     -H "Authorization: ${TOKEN}" \
-     -XPOST https://quay.io/cnr/api/v1/packages/${NAMESPACE}/${REPOSITORY} -d '
-{
-    "blob": "'"${BLOB}"'",
-    "release": "'"${RELEASE}"'",
-    "media_type": "helm"
-}'
-```
+## Using Operator Manifest in an `OperatorSource`
 
-If you have saved the manifest into `myapp.yaml` and want to push it to `myoperators/myapp:1.0.0` then the variables will have the following values.
-```bash
-FILENAME="myapp"
-NAMESPACE="myoperators"
-REPOSITORY="myapp"
-RELEASE="1.0.0"
-```
+*The following requires that Marketplace be deployed in your cluster*
+
+Once the operator manifest is pushed to `quay.io` You can also refer to your operator manifest in an `OperatorSource` to add your offerings to Marketplace. An example `OperatorSource` is provided [here](examples/operatorsource.cr.yaml).
+
+An `OperatorSource` must specify the `registryNamespace` for the operator manifest, and set the `name` and `namespace` for creating the `OperatorSource` on your cluster.
```

Comment:
Please map this to the input variables of the script.

Reply:
To address some of @tkashem's concerns, this information can be moved to the main README which anyway needs a link to this doc.